File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        HashMap<Character, Integer> map = new HashMap<>();
        int longestSubStrLength = 0;

        for(int left = 0, right = 0; right < s.length(); right++) {
            Character ch = s.charAt(right);
            map.put(ch, map.getOrDefault(ch, 0) + 1 );
            while(map.get(ch) > 1) {
                Character leftCh = s.charAt(left);
                map.put(leftCh, map.getOrDefault(leftCh, 0) - 1 );
                left++;
            }
            longestSubStrLength = Math.max(longestSubStrLength, right - left + 1);
        }
        return longestSubStrLength;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-4.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        map = [0] * 26
        

        if len(s) != len(t):
            return False
        
        for i in range(len(s)):
            map[ord(s[i]) - ord('a')] += 1
            map[ord(t[i]) - ord('a') ] -= 1
        return True if max(map)==0 else False